tests/test_runlib.py

```diff
@@ -642,6 +662,16 @@ def test_create_metadata_with_expected_product(self):
     self.assertEqual(list(link.signed.products.keys()), [self.test_product])
     os.remove(self.link_name)
 
+  def test_create_metadata_with_expected_product_in_target_directory(self):
+    """Test record stop records expected product in target directory """
+    in_toto_record_start(self.step_name, [], self.key)
+    in_toto_record_stop(self.step_name, [self.test_product], self.key,
+        metadata_directory=self.metadata_directory)
+    link_path = os.path.join(self.metadata_directory, self.link_name)
+    link = Metablock.load(link_path)
+    self.assertEqual(list(link.signed.products.keys()), [self.test_product])
+    os.remove(self.link_name)
```

Comment:
+1 on clean up! I think this will be easier with a teardown method perhaps, and if we're using a newly created temporary directory.

Reply:
I am not sure about using a teardown method. Other test functions in this class all use `os.remove()` and I just follow this pattern.

Reply:
That makes sense, really this only applies if we're creating a class for `metadata_directory` specific tests. Depends on how many we have, and how many can use the same directory that we create (so that we can do it once for the class and reuse). I think we can work out the rest of it and reorganize?

Reply:
IMO you don't need to wrap this in a try/finally block, it makes it a tiny bit harder to read and the whole test case is run in a temporary directory, which gets removed on tearDownClass anyway (see TmpDirMixin).